src/goto-instrument/code_contracts.cpp

```diff
@@ -563,7 +568,8 @@ void code_contractst::instrument_assign_statement(
   alias_assertion.add(goto_programt::make_assertion(
     assigns_clause.alias_expression(lhs),
     instruction_iterator->source_location));
-
+  alias_assertion.instructions.back().source_location.set_comment(
+    "Check assigns clause");
```

Comment:
I think the `Check ensures clause` and `Check requires clause` messages are fine, because each function currently can have at most one ENSURES clause and at most one REQUIRES clause.

But multiple `Check assigns clause` assertions seem less useful because it's not clear which variable each assertion is looking at.

There might be a better way to do this, but this could be one way:
```
alias_assertion.instructions.back().source_location.set_comment(
    "Check that " + lhs.pretty(0) + " is assignable");
```

@tautschnig Do you have any suggestions on what we should use here to `string`-ify `lhs` here? Is `pretty` the right thing to use?

Reply:
As we discussed offline, `pretty` seems to generate a verbose string (with lots of debug info).

Based on a quick search through the code base, I think `from_expr` (from `#include <langapi/language_util.h>) could be the used here, e.g. https://github.com/diffblue/cbmc/blob/1ab5de1ac0893c7ca94ffe5463e2c7c4ee781266/src/goto-programs/builtin_functions.cpp#L752-L753

Reply:
@SaswatPadhi I also tried `from_expr`, but at the end it also uses pretty and generates the same verbose strings.
https://github.com/diffblue/cbmc/blob/1ab5de1ac0893c7ca94ffe5463e2c7c4ee781266/src/langapi/language.cpp#L32-L39

I'm now converting `lhs` to a symbol and extracting the identifier, which results to the following message
```console
main.c function bar
[bar.1] line 4 Check that bar::y is assignable: SUCCESS
```

Reply:
The above `from_expr` definition looks suspicious, I don't think that's the one being called from the `set_comment` example I showed.

The `from_expr` call should return a `std::string` but the definition you posted above returns a `bool`. Could you please double check?

It's being used within `set_comment`, so if it prints a verbose output then that should break the CBMC output ...

Reply:
I believe this is the right definition:

https://github.com/diffblue/cbmc/blob/48893287099cb5780302fe9dc415eb6888354fd6/src/langapi/language_util.cpp#L20-L31

Reply:
Oh may be you're right, `p->from_expr` probably resolves to the definition you mentioned ...

Reply:
Okay, this should be called, not the abstract definition that you posted above.

https://github.com/diffblue/cbmc/blob/1ab5de1ac0893c7ca94ffe5463e2c7c4ee781266/src/ansi-c/ansi_c_language.cpp#L146-L149

This does call `expr2c`.

Basically in this case the `get_language_from_identifier` should resolve to C language, not the abstract `languaget`.

Reply:
Thanks for the tip! It worked perfectly.